warehouse/query-core/src/test/java/datawave/query/jexl/visitors/QueryPruningVisitorTest.java

```diff
@@ -0,0 +1,291 @@
+        
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        JexlNode reduced = QueryPruningVisitor.reduce(script);
+        
+        Assert.assertEquals(query, JexlStringBuildingVisitor.buildQuery(reduced));
+    }
+    
+    @Test
+    public void combinationTest() throws ParseException {
+        String query = "F(_NOFIELD_) == 'x'";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.UNKNOWN, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void andTrueTest() throws ParseException {
+        String query = "true && !(_NOFIELD_ == 'z')";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.TRUE, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void orFalseTest() throws ParseException {
+        String query = "false || _NOFIELD_ == 'z'";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.FALSE, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void notUnknownTest() throws ParseException {
+        String query = "FIELD != 'a'";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.UNKNOWN, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void notNoFieldTest() throws ParseException {
+        String query = "_NOFIELD_ != 'a'";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.FALSE, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void notFalseTest() throws ParseException {
+        String query = "!false";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.TRUE, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void notTrueTest() throws ParseException {
+        String query = "!true";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.FALSE, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void methodTest() throws ParseException {
+        String query = "x('a')";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.UNKNOWN, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
+    public void functionTest() throws ParseException {
+        String query = "filter:isNull(FIELD)";
+        ASTJexlScript script = JexlASTHelper.parseJexlQuery(query);
+        Assert.assertEquals(QueryPruningVisitor.TRUTH_STATE.UNKNOWN, QueryPruningVisitor.getState(script));
+    }
+    
+    @Test
```

Comment:
Consider a bounded range test? A un-indexed field test?

Reply:
This class (QueryPruningVisitor) doesn't know the difference between indexed/unindexed, it is either a _NOFIELD_ or not. Since a bounded range is really a concept instead of a structural construct I'm not sure that has a place in this class either.

Reply:
I think we have some integration tests that exercise bounded ranges, but I will need to try to confirm that.